pkgs/by-name/gu/gupax/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  autoPatchelfHook,
+  darwin,
+  fetchFromGitHub,
+  git,
+  lib,
+  libglvnd,
+  libxkbcommon,
+  openssl,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  vulkan-loader,
+  wayland,
+  xorg,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "gupax";
+  version = "1.3.8";
+
+  src = fetchFromGitHub {
+    owner = "hinto-janai";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-duDqlCUAMYXM+M3Ifzwshlkr80PcqqXE/G4zih/uwA0=";
+    leaveDotGit = true; # git command in build.rs
+  };
+
+  cargoHash = "sha256-a9Q0/AA/kJCcZl3maROuwUH9/tBGDZdxqBYSEIXr6EA=";
+
+  cargoBuildFlags = [
+    # disables updates but also uses /usr/bin paths
+    # p2pool and xmrig will need to be symlinked anyways
+    "--features distro"
+  ];
+
+  checkFlags = [
+    # requires filesystem write
+    "--skip disk::test::create_and_serde_gupax_p2pool_api"
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    git
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      openssl
+      stdenv.cc.cc.libgcc or null # libgcc_s.so
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+      darwin.apple_sdk.frameworks.ApplicationServices
+      darwin.apple_sdk.frameworks.Carbon
+      darwin.apple_sdk.frameworks.CoreData
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreGraphics
+      darwin.apple_sdk.frameworks.CoreServices
+      darwin.apple_sdk.frameworks.CoreVideo
+      darwin.apple_sdk.frameworks.Foundation
+      darwin.apple_sdk.frameworks.IOKit
+      darwin.apple_sdk.frameworks.Metal
+      darwin.apple_sdk.frameworks.OpenGL
+      darwin.apple_sdk.frameworks.QuartzCore
+      darwin.apple_sdk.frameworks.Security
+    ];
+
+  runtimeDependencies = [
+    libglvnd # libEGL.so
+    libxkbcommon # libxkbcommon.so
+    vulkan-loader # libvulkan.so
+    wayland # libwayland-client.so
+    xorg.libX11 # libX11.so
+    xorg.libXcursor # libXcursor.so
+    xorg.libXi # libXi.so
```

Comment:
Are these linux-specific dependencies?

Reply:
They are. I am not sure if other platforms use display manager libraries but I doubt it. Can put an optional here if the Darwin build work but that is currently not the case.